File: TF14. LISTS - EXERCISES/P02.ChangeList.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace P02.ChangeList
{
    class Program
    {
        static void Main(string[] args)
        {
               List<int> input = Console.ReadLine()
              .Split()
              .Select(int.Parse)
              .ToList();

            while (true)
            {
                string[] command = Console.ReadLine().Split();

                if (command[0] == "Delete")
                {
                    input.RemoveAll(x => x == int.Parse(command[1]));
                }
                else if (command[0] == "Insert")
                {
                    input.Insert(int.Parse(command[2]), int.Parse(command[1]));
                }
                else if (command[0] == "Even")
                {
                    for (int i = 0; i < input.Count; i++)
                    {
                        if (input[i] % 2 == 0)
                        {
                            Console.Write($"{input[i]} ");
                        }
                    }
                    Console.WriteLine();
                    break;
                }
                else if (command[0] == "Odd")
                {
                    for (int i = 0; i < input.Count; i++)
                    {
                        if (input[i] % 2 != 0)
                        {
                            Console.Write($"{input[i]} ");
                        }
                    }
                    Console.WriteLine();
                    break;
                }
            }
        }
    }
}

File: TF14. LISTS - EXERCISES (Java)/P02_ChangeList.java
import java.io.Console;
import java.util.*;
import java.util.stream.Collectors;

public class P02_ChangeList {
    public static void main(String[] args) {

        Scanner in = new Scanner(System.in);

        List<Integer> input = Arrays.stream(in.nextLine().split(" "))
                .map(Integer::parseInt)
                .collect(Collectors.toList());

        while (true) {
            String[] command = in.nextLine().split(" ");

            if (command[0].equals("Delete")) {

                int removed = Integer.parseInt(command[1]);

                input.removeAll(Collections.singleton(removed));

            } else if (command[0].equals("Insert")) {

                input.add(Integer.parseInt(command[2]), Integer.parseInt(command[1]));

            } else if (command[0].equals("Even")) {

                for (Integer anInput : input) {

                    if (anInput % 2 == 0) {

                        System.out.printf("%d ", anInput);
                    }
                }
                System.out.println();
                break;

            } else if (command[0].equals("Odd")) {

                for (Integer anInput : input) {

                    if (anInput % 2 != 0) {

                        System.out.printf("%d ", anInput);
                    }
                }
                System.out.println();
                break;
            }
        }
    }
}
